doc.js

```diff
@@ -394,6 +394,26 @@ define('doc', ['event'], function(event) {
 				} else {
 					return;
 				}
+			},
+
+			'insertBefore' : function(elements) {
+				if(typeof elements == 'string'){
+					elements = query(search(document, elements));
+				}
+
+				elements.each(function(el) {
+					el.insertAdjacentHTML('beforebegin', this.els[0].outerHTML);
```

Comment:
As discussed privately with @erikatakahara, inserting before/after elements without parent won't be supported. It's still possible to create a complex structure of elements before inserting them into the DOM by creating a root for them, which is the most common use case.

@luiz